app/full_text_search.py

```diff
@@ -171,10 +214,15 @@ def create_document(record_id, repo, **kwargs):
     fields.append(appengine_search.TextField(name='record_id', value=record_id))
 
     # Add name and location romanized by unidecode
+    name_fields = ['given_name', 'full_name', 'family_name', 'alternate_names']
     for field in kwargs:
         romanized_value = script_variant.romanize_word(kwargs[field])
-        fields.append(
-            appengine_search.TextField(name=field, value=romanized_value))
+
+        if field in name_fields:
```

Comment:
Add a comment why you call create_fields_for_rank for name_fields, but not for other fields.